geonode/base/views.py

```diff
@@ -43,6 +43,72 @@
 from geonode.base.forms import BatchEditForm, OwnerRightsRequestForm
 from geonode.base.forms import CuratedThumbnailForm
 from geonode.notifications_helper import send_notification
+from geonode.base.forms import UserAndGroupPermissionsForm
+from django.contrib.auth import get_user_model
+from geonode.tasks.tasks import set_permissions
+from django.contrib.auth.models import Group
+
+
+def user_and_group_permission(request, model):
+    if not request.user.is_superuser:
+        raise PermissionDenied
+
+    model_mapper = {
+        "profile": get_user_model(),
+        "group": Group
+    }
+
+    model_class = model_mapper[model]
+
+    ids = request.POST.get("ids")
+    if "cancel" in request.POST or not ids:
+        return HttpResponseRedirect(
+            '/admin/{}/{}/'.format(model_class._meta.app_label, model)
```

Comment:
reverse?

Reply:
@petrus7, the URLs being generated in the above line are Django admin URLs for layers, documents, and maps. Sincerely speaking, I do not know where to find the namespaces for the routes. I tried using the formats described [here](https://stackoverflow.com/questions/694477/getting-django-admin-url-for-an-object), but they do not seem to work too. Any ideas? I also do not think it is a very good idea to change lines of code not related to the current issue (that is, lines that I did not change).

Reply:
agreed